src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -379,6 +394,69 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
     return found;
   }
 
+  /**
+   * @inheritDoc
+   */
+  getFeatures(pixel) {
+    return new Promise(function(resolve, reject) {
+      const layer = /** @type {import("../../layer/VectorTile.js").default} */ (this.getLayer());
+      const source = layer.getSource();
+      const projection = this.renderedProjection;
+      const projectionExtent = projection.getExtent();
+      const resolution = this.renderedResolution;
+      const pixelRatio = this.renderedPixelRatio;
+      const tileGrid = source.getTileGridForProjection(projection);
+      const sourceTileGrid = source.getTileGrid();
+      const coordinate = applyTransform(this.renderedPixelToCoordinateTransform_, pixel.slice());
+      const tileCoord = tileGrid.getTileCoordForCoordAndResolution(coordinate, resolution);
+      let sourceTile;
+      for (let i = 0, ii = this.renderedTiles.length; i < ii; ++i) {
+        if (tileCoord.toString() === this.renderedTiles[i].tileCoord.toString()) {
+          const tile = this.renderedTiles[i];
+          if (tile.getState() === TileState.LOADED && tile.hifi) {
+            const extent = tileGrid.getTileCoordExtent(tileCoord);
+            if (source.getWrapX() && projection.canWrapX() && !containsExtent(projectionExtent, extent)) {
+              const worldWidth = getWidth(projectionExtent);
+              const worldsAway = Math.floor((coordinate[0] - projectionExtent[0]) / worldWidth);
+              coordinate[0] -= (worldsAway * worldWidth);
+            }
+            const sourceTiles = source.getSourceTiles(pixelRatio, projection, tile);
+            const sourceTileCoord = sourceTileGrid.getTileCoordForCoordAndResolution(coordinate, resolution);
+            for (let j = 0, jj = sourceTiles.length; j < jj; ++j) {
+              if (sourceTileCoord.toString() === sourceTiles[j].tileCoord.toString()) {
+                sourceTile = sourceTiles[j];
+                break;
+              }
+            }
+          }
+          break;
+        }
+      }
+      const corner = getTopLeft(tileGrid.getTileCoordExtent(sourceTile.tileCoord));
```

Comment:
I'm getting the following error on this line:
```
Uncaught (in promise) TypeError: Cannot read property 'tileCoord' of undefined
    at e.<anonymous> (VectorTileLayer.js:435)
    at new Promise (<anonymous>)
    at e.getFeatures (VectorTileLayer.js:401)
    at e.getFeatures (Layer.js:201)
    at e.getFeatures (BaseVector.js:153)
    at e.getFeatures (VectorTile.js:134)
    at e.<anonymous> (vector-tile-selection.js:48)
    at e.dispatchEvent (Target.js:110)
    at e.handleMapBrowserEvent (PluggableMap.js:946)
    at e.dispatchEvent (Target.js:110)
```

This happens randomly when clicking around the map in [the selection example](https://399-4974939-gh.circle-artifacts.com/0/examples/vector-tile-selection.html). Not very easy to reproduce, let me know if you need more precise instructions.

Reply:
I was able to reproduce this - it was due to tiles that were not loaded yet, when we should actually resolve with an empty array.